io/disk/arcconf/arcconf_migration.py

```diff
@@ -0,0 +1,246 @@
+                    self.logical_print()
+                    time.sleep(int(self.migration_sleep))
+
+            # Migration tests
+            else:
+                self.log.info("Migrating from RAID %s to %s" %
+                              (self.initial_raid, migrate))
+                cmd = "echo y | arcconf MODIFY %s FROM %s TO %s %s %s" % \
+                      (self.crtl_no, self.logicaldrive,
+                       self.disk_size, migrate, disk_val2)
+                self.check_pass(cmd, "Failed to run %s" % cmd)
+                time.sleep(int(self.migration_sleep))
+                self.logical_print()
+
+            if self.cmdop_list("OS"):
+                cmd = "echo y | arcconf delete %s logicaldrive \
+                       %s" % (self.crtl_no, self.logicaldrive)
+                self.check_pass(cmd, "Logical drive deletion failed")
+
+            time.sleep(int(self.migration_sleep))
+
+            # restore global spare only after logical drive delete
+            if self.spare_drive != "":
+                cmd = "echo y | arcconf setstate %s device %s %s RDY " % \
+                       (self.crtl_no, self.channel_no, self.spare_drive)
+                self.check_pass(cmd, "Failed to run %s" % cmd)
+
+    def logical_print(self):
+        """
+        Function to print logical drive status
+        """
+        cmd = "arcconf getconfig %s LD" % (self.crtl_no)
+        self.check_pass(cmd, "Logical drive getconfig failed")
+
+    def cmdop_list(self, cmd):
+        """
+        Function returns the output of a command
+        """
+        if cmd == 'OS':
+            cmd = "lsscsi  | grep LogicalDrv | awk \'{print $7}\'"
+        val = process.run(cmd, shell=True, ignore_status=True,
+                          allow_output_check='stdout')
+        if val.exit_status:
+            self.fail("cmd %s Failed" % (cmd))
+        return val.stdout.rstrip()
+
+    @classmethod
+    def comp(cls, list1, list2):
+        """
+        compares two lists, if match found returns 0
+        """
+        for val in list1:
+            if val in list2:
+                return 0
+        return 1
+
+    def check_pass(self, cmd, errmsg):
+        """
+        Function to check if the cmd is successful or not
+        """
+        if process.system(cmd, timeout=3200, ignore_status=True,
+                          shell=True) != 0:
+            self.fail(errmsg)
+
+    def tearDown(self):
+        """
+        Delete logical device
+        """
+        drive = self.cmdop_list("OS")
+
+        if drive != "":
+            # Delete a logical drive only if it is not OS drive
+            cmd = "df -h /boot | grep %s" % drive
+            if process.system(cmd, timeout=300, ignore_status=True,
+                              shell=True) != 0:
+                cmd = "echo y | arcconf delete %s logicaldrive %s" % \
+                      (self.crtl_no, self.logicaldrive)
+                self.check_pass(cmd, "Failed to cleanup Logical drive")
```

Comment:
@narasimhan-v  tried your suggestion. the code seems to be running, the fix will be in v3